src/SFML/Graphics/Shader.cpp

```diff
@@ -103,20 +103,20 @@ bool getFileContents(const std::filesystem::path& filename, std::vector<char>& b
 // Read the contents of a stream into an array of char
 bool getStreamContents(sf::InputStream& stream, std::vector<char>& buffer)
 {
-    bool               success = false;
-    const std::int64_t size    = stream.getSize();
-    if (size > 0)
+    bool       success = false;
+    const auto size    = stream.getSize();
+    if (size.has_value() && *size > 0)
```

Comment:
What about this:
```suggestion
    if (size > 0)
```

Reply:
<img width="793" alt="Screenshot 2024-04-30 at 12 38 42 PM" src="https://github.com/SFML/SFML/assets/39244355/e9758c6c-dd39-4896-bef8-39531124d699">

This is valid but we have to be careful because `operator<` has the opposite behavior and will return `true` if `size` is null.